File: frontend/api/req/CreateTaskReq.ts
import { z } from "zod";

const CreateTaskReqSchema = z.object({
  title: z.string(),
  description: z.optional(z.string().nullable()),
  dueDate: z.optional(z.string().nullable()),
});

type CreateTaskReq = z.infer<typeof CreateTaskReqSchema>;

export { CreateTaskReqSchema };
export type { CreateTaskReq };

File: backend/Server/Contracts/Requests/CreateTaskReq.cs
using Server.Contracts.Entities;

namespace Server.Contracts.Requests;

public class CreateTaskReq
{
    public string Title { get; set; } = default!;
    public string? Description { get; set; }
    public DateOnly? DueDate { get; set; }
    public TaskPriorityEnum Priority { get; set; }
}